test/http/http-client.test.js

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const should = require('should');
+
+const { importFile, getTestOptions } = require('../testutils');
+const MockRequest = require('../mock-request');
+
+const HttpClient = importFile('http/http-client');
+const HttpRequest = importFile('http/http-request');
+const DirectBinaryUploadOptions = importFile('direct-binary-upload-options');
+const HttpResult = importFile('http-result');
+
+describe('HTTP Client Tests', function() {
+    let httpResult;
+    let options;
+    let uploadOptions;
+    let httpClient;
+
+    beforeEach(function() {
+        MockRequest.reset();
+        options = getTestOptions();
+        uploadOptions = new DirectBinaryUploadOptions()
+            .withHttpRetryDelay(10);
+        httpClient = new HttpClient(options, uploadOptions);
+        httpResult = new HttpResult(options, uploadOptions);
+    });
+
+    function createHttpRequest(url) {
+        return new HttpRequest(options, url);
+    }
+
+    it('http request submit smoke test', async function() {
+        MockRequest.onGet('http://myunittest.com').reply(200, 'hello world!');
```

Comment:
I would choose another url, because `http://myunittest.com` actually exists!
```suggestion
        MockRequest.onGet('http://adobe-aem-upload-unit-test.com').reply(200, 'hello world!');
```

Reply:
Good point, I replaced the URL in all applicable tests.